src/bio/bio_context.c

```diff
@@ -574,22 +615,38 @@ bio_blob_open(struct bio_io_context *ctxt, bool async)
 	ctxt->bic_io_unit = spdk_bs_get_io_unit_size(bbs->bb_bs);
 	D_ASSERT(ctxt->bic_io_unit > 0 && ctxt->bic_io_unit <= BIO_DMA_PAGE_SZ);
 
-	/*
-	 * Query per-server metadata to get blobID for this pool:target
-	 */
-	rc = smd_pool_get_blob(ctxt->bic_pool_id, xs_ctxt->bxc_tgt_id,
-			       &blob_id);
-	if (rc != 0) {
-		D_ERROR("Failed to find blobID for xs:%p, pool:"DF_UUID", tgt:%d\n",
-			xs_ctxt, DP_UUID(ctxt->bic_pool_id), xs_ctxt->bxc_tgt_id);
-		return -DER_NONEXIST;
-	}
-
 	bma = blob_msg_arg_alloc();
 	if (bma == NULL)
 		return -DER_NOMEM;
 	ba = &bma->bma_cp_arg;
 
+	if (open_blobid == SPDK_BLOBID_INVALID) {
+		/*
+		 * Query per-server metadata to get blobID for this pool:target
+		 */
+		if (!is_sys || !bio_is_meta_on_ssd_configured()) {
+			rc = smd_pool_get_blob(ctxt->bic_pool_id, xs_ctxt->bxc_tgt_id,
+					       &blob_id);
+			if (rc != 0) {
+				D_ERROR("Failed to find blobID for xs:%p, pool:"DF_UUID", tgt:%d\n",
+					xs_ctxt, DP_UUID(ctxt->bic_pool_id), xs_ctxt->bxc_tgt_id);
+				rc = -DER_NONEXIST;
+				goto out_free;
+			}
+		} else {
+			ba->bca_inflights = 1;
+			spdk_bs_get_super(bbs->bb_bs, get_super_cb, bma);
```

Comment:
Minor: spdk_bs_get_super() is a synchronous call, it just get the blob ID from spdk_blob_store->super_blob.